### PR TITLE
feat(agent): host address env & host scheme

### PR DESCRIPTION
**Pull request by robot9706**
- Add `HOST_ADDRESS` environment to define the server address when installing an agent.
- Add the option to define the scheme of the address, if it's `http` an insecure connection will be used in any environment.

### CODE REVIEW

File: golang/internal/config/config.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type CommonConfiguration struct {
 	ImportContainerImage     string        `yaml:"importContainerImage"     env:"IMPORT_CONTAINER_IMAGE"      env-default:"rclone/rclone:1.57.0"` //nolint:lll
 	RootDomain               string        `yaml:"rootDomain"               env:"ROOT_DOMAIN"                 env-default:""`
 	ReadHeaderTimeout        time.Duration `yaml:"readHeaderTimeout"        env:"READ_HEADER_TIMEOUT"         env-default:"15s"`
+	HostAddress              string        `yaml:"hostAddress"              env:"HOST_ADDRESS"                env-default:""`
```
Comment (polaroi8d) [inline golang/internal/config/config.go:40]: Could we add a comment in here that we are using this in darklens only?

**Comment by nandor-magyar** [comment]
The original version, agent gets the issuer URL and uses that, shouldn't we check at least the the basedomain in the issuer and in the JWT?

**Comment by robot9706** [comment]
@nandor-magyar The problem is that dyo (and darklens) has the agent address hardcoded in an env, so if your agent needs to access the server on a different route you need to modify that env so the JWT has the correct issuer. Let's say one agent uses `host.docker.internal` another uses a local IP and a third uses a public domain, you'd need to modify that env 3 times just to be able to install 3 agents. So comparing the HOST_ADDRESS and JWT issuer won't work I think.

File: golang/internal/config/config.go (hunk 1 above)
Comment (robot9706) [reply golang/internal/config/config.go:40]: We could also use it in dyo, but sure :)

**Comment by polaroi8d** [comment]
Based on an internal discussion, the feature will not be included in the agent.

**Pull request closed without merging**